src/Microsoft.OpenApi.OData.Reader/Common/EdmModelHelper.cs

```diff
@@ -141,15 +143,29 @@ internal static string GenerateComplexPropertyPathOperationId(ODataPath path, st
                 items.Add(Utils.UpperFirstChar(lastSegment?.Identifier));
             }
 
-            return string.Join(".", items);
+            return GenerateNavigationPropertyPathOperationId(items);
+        }
+
+        /// <summary>
+        /// Generates a navigation property operation id from a list of string values.
+        /// </summary>
+        /// <param name="items">The list of string values.</param>
+        /// <returns>The generated navigation property operation id.</returns>
+        private static string GenerateNavigationPropertyPathOperationId(IList<string> items)
+        {
+            if (!items.Any())
+                return null;
+
+            return string.Join(".", items).Replace(".-", "-"); // Format any hashed value appropriately (this will be the last value)
```

Comment:
```suggestion
            return string.Join(".", items).Replace(".-", "-", StringComparison.OrdinalIgnoreCase); // Format any hashed value appropriately (this will be the last value)
```